### ISSUE TITLE
Get list of missing songs

### ISSUE THREAD

**Issue by aNerdgirl** [rnagabhyrava/plex-playlist-sync#15]
An additional question. Is it possible to get the list of songs that are not available in Plex (just to be aware of the difference between Spotify and Plex).
I tried to add a command to get_available_plex_tracks() (write track and artist to a .csv file, if except IndexError in line 235 and except BadRequest in line 211 ), but it seems to not pick-up all songs. Am I missing something?

**Comment by rnagabhyrava** [comment]
Yes, its possible to get missing songs and you're looking at the right place. You could do something like below to write out the missing tracks, you may also want to pass in some identifiers(like playlist name) to the function that way you know which playlist the missing songs belong to or the missing_tracks.csv will be overwritten for each playlist. Currently, the script doesn't write any thing out so no specific mount is needed in the docker image. To run this with docker you may want to add appropriate permissions (PUID, GUID) and a mount location to write to.

```diff
import pandas as pd
def get_available_plex_tracks(plex: PlexServer, trackZip: List) -> List:
    """For the given spotify track names returns a list of plex.audio.track objects
        - Empty list if none of the tracks are found in Plex
    Args:
        plex (PlexServer): A configured PlexServer instance
        trackNames (List): List of track names
    Returns:
        List: of track objects
    """
    musicTracks = []
+   missingTracks = []
    for track, artist, album in trackZip:
        try:
            search = plex.search(track, mediatype='track', limit=5)
        except BadRequest:
            logging.info("failed to search %s on plex", track)
            search = []
        if not search:
            logging.info("retrying search for %s", track)
            try:
                search = plex.search(
                    track.split('(')[0], mediatype='track', limit=5
                )
                logging.info("search for %s successful", track)
            except BadRequest:
                logging.info("unable to query %s on plex", track)
                search = []

        if search:
            for s in search:
                try:
                    artistSimilarity = SequenceMatcher(
                        None, s.artist().title.lower(), artist.lower()
                    ).quick_ratio()

                    if s.artist().title.lower() == artist.lower() or artistSimilarity >= 0.8:
                        musicTracks.extend(s)
                        break

                    albumSimilarity = SequenceMatcher(
                        None, s.album().title.lower(), album.lower()
                    ).quick_ratio()

                    if s.album().title.lower() == album.lower() or albumSimilarity >= 0.8:
                        musicTracks.extend(s)
                        break

                except IndexError:
                    logging.info(
                        "Looks like plex mismatched the search for %s, retrying with next result", track)
        
+       if not search:
+           missingTracks.append([track, artist, album])

+   pd.DataFrame(missingTracks).to_csv('missing_tracks.csv', index=False, columns=["track", "artist", "album"])
    return musicTracks
```

**Comment by rnagabhyrava** [comment]
Implemented this in 2016c17b57fb2f8280b4eeadbd0f4c2fa5776e43
